packages/gestalt/src/Toast.js

```diff
@@ -59,82 +103,90 @@ type Props = {|
  * ![Toast light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Toast.spec.mjs-snapshots/Toast-chromium-darwin.png)
  */
 export default function Toast({
+  _dangerouslySetPrimaryAction,
+  dismissButton,
+  helperLink,
   primaryAction,
   text,
   thumbnail,
-  thumbnailShape = 'square',
-  variant = 'default',
-  _dangerouslySetPrimaryAction,
+  type,
 }: Props): Node {
   const { name: colorSchemeName } = useColorScheme();
   const isDarkMode = colorSchemeName === 'darkMode';
-  const isErrorVariant = variant === 'error';
 
   const responsiveMinWidth = useResponsiveMinWidth();
   const isMobileWidth = responsiveMinWidth === 'xs';
 
-  let containerColor = isDarkMode ? 'light' : 'dark';
-  let textColor = isDarkMode ? 'dark' : 'light';
-
   let textElement: Element<'span'> | string;
 
   if (typeof text === 'string') {
     textElement = text;
   }
 
   // If `text` is a Text component, we need to override any text colors within to ensure they all match
-  if (typeof text !== 'string' && Children.only(text).type.displayName === 'Text') {
+  const isTextNode = typeof text !== 'string' && Children.only(text).type.displayName === 'Text';
+
+  if (isTextNode) {
     let textColorOverrideStyles = isDarkMode
       ? styles.textColorOverrideDark
       : styles.textColorOverrideLight;
 
-    if (isErrorVariant) {
+    if (type === 'error') {
       textColorOverrideStyles = styles.textErrorColorOverrideLight;
```

Comment:
Not quite related to these changes, but do we need a dark mode error override here as well? (apologies if I've asked this before — though that would indicate that an explanatory comment here would be helpful)